Add comment section

dtalton3 opened the pull request:
Created a comment section for users to post their opinion of my webpage... or just me in general.
 
- Ability to do post request
- Display user comments on webpage

dtalton3 committed:
Add dependency for Servlet

dtalton3 committed:
Modify portfolio to use fetch requests for Fun Fact Generator

dtalton3 committed:
Resolve comments and add the forgotten script.js

dtalton3 committed:
Add ability to do post request.

dtalton3 committed:
Design comment section for portfolio UI.

dtalton3 committed:
Allow webpage to display user's comments with showComments() method.

dtalton3 committed:
Format the comment section of the webpage.

octopus1000 reviewed (changes requested):
Can you please merge "Week 3 Walkthrough Part 2" to master and  rebase this branch from merged master? ---> This way changes in previous pull request won't show up here.

Also, think about giving a more meaning full name for the pull request. Eg. "Adding comment section".

dtalton3 committed:
Merge branch 'master' into postrequest

dtalton3 commented:
> Can you please merge "Week 3 Walkthrough Part 2" to master and rebase this branch from merged master? ---> This way changes in previous pull request won't show up here.
> 
> Also, think about giving a more meaning full name for the pull request. Eg. "Adding comment section".

Done. Rebased, merged and changed the name of the pull request.

addarsh-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -32,16 +32,34 @@ function funFactGenerator() {
-    const factGetter = document.getElementById('factgetter');
-    factGetter.innerText = fact;
+    // Starts post request using user's input.
+    var init = {method: "POST",
```
you probably don't need this variable. Just create the JSON object when passing in the request argument in the next line.

dtalton3 replied:
Alright. I'll do that

addarsh-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -32,16 +32,34 @@ function funFactGenerator() {
 
+    // Fetches server comment data and writes it to the page.
+    fetch(request).then(comments => comments.json()).then(jsoncomments => {
+        document.getElementById("thecomments").innerHTML += "<p>" + 
```
do you need tags inside of an element? Not sure how this looks in the UI.

dtalton3 replied:
Yeah I need the tags so that each comment is on their own line. I could have used "\n" likely but I wanted to leave myself the option to format those comments a little better in CSS later.

addarsh-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -32,16 +32,34 @@ function funFactGenerator() {
+        const fact = jsonfacts[Math.floor(Math.random() * jsonfacts.length)];
+
+        // Add it to the page.
+        const factGetter = document.getElementById('factgetter');
```
You don't need the factGetter var, try chaining with the next line.

dtalton3 replied:
Will do .

addarsh-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -32,16 +32,34 @@ function funFactGenerator() {
+}
+
+/**
+ * Adds user's comment to the page.
```
is this description still valid? You may want to explain that it retrieves all comments as well?

dtalton3 replied:
For now this description is still valid... it only adds the last comment indexed of the comments arraylist (which is the one the user just posted) *line 60/61*... it wont be valid once I implement the datastore though so I'll be sure to update that comment when I implement that.

dtalton3 committed:
Resolve comments.

dtalton3 commented:
Resolved comments. 

- Got rid of unnecessary vars
